unit-tests/style_test.py

```diff
@@ -79,7 +80,8 @@ def test_constructor(self, initials):
         style.
         """
         unit_under_test = TestStyle._StyleHarness(initials[0])
-        assert unit_under_test.list_rules() == initials[1]
+        assert [rule.__class__.__name__
```

Comment:
Should the objects under `rule` have an attribute to store some sort of identifying name so that you don't need to pull it from the `__` attributes?

Reply:
An interesting question. If this were the main code then it would not even be a question - use of internal attributes like that is contra-indicated. However test code has different requirements and different concerns. Simplicity and independence are the keys to test code. Adding things to the main code just for testing is also contra-indicated.

In particular, is it worth implementing something to provide the same functionality that is all ready provided by the language? I return it to you for a decission.